src/cryptography/hazmat/backends/openssl/x509.py

```diff
@@ -151,3 +152,67 @@ def signature_hash_algorithm(self):
             raise UnsupportedAlgorithm(
                 "Signature algorithm OID:{0} not recognized".format(oid)
             )
+
+    @property
+    def extensions(self):
+        # TODO: remove this when we finish landing support for extensions.
+        warnings.warn(
+            "Extension support is not fully implemented. Unsupported "
+            "extensions are IGNORED at this time."
+        )
+        extensions = []
+        extcount = self._backend._lib.X509_get_ext_count(self._x509)
+        if extcount > 0:
+            bc = self._get_extension_by_nid(
+                self._backend._lib.NID_basic_constraints, "basicConstraints"
+            )
+            if bc:
+                ext = self._build_basic_constraints(bc)
+                extensions.append(ext)
```

Comment:
I think it would make more sense to iterate through the extensions in order, rather than searching for specific ones in list.